src/common/outputService/outputService.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/* eslint-disable no-console, class-methods-use-this */
+
+import { green, red } from 'chalk';
+import { AsyncOperationResult } from '../types';
+
+/**
+ * Service class to print the output
+ *
+ * @author JuanStenghele-sf
+ */
+export abstract class OutputService {
```

Comment:
I think this class hierarchy should be broken up a bit more.

abstract class OutputService ==> Only things that are used by all commands (maybe nothing now)
abstract class AorOutputService extends OutputService ==> interfaces that deal with an AOR
interface ResumeOutputService.  Generic things needed by resume logic.  Like printProgressSummary
class ResumeCommandOutputService extends AorOutputService implements ResumeOutputService 
interface DeployOutputService.  Generic things about deployment
class DeployCommandOutputService extends AorOutputService implements DeployOutputService, ResumeOutPutService

Reply:
printProgressSummary  is not just for the ResumeOutputService, we also use it for the deploy itself. What would you put in ResumeOutputService?